Keywords/commonUtils/WebTableUtils.groovy

```diff
@@ -1,253 +1,285 @@
+		WebElement tableElement = WebUI.findWebElement(webTableObject, 0)
+		List<String> tableData = new ArrayList<>();
+		try {
+			String rowData = null;
+			int count = 0
+
+			// Get all rows
+			List<WebElement> rows = tableElement.findElements(By.tagName("tr"));
+
+			//Get all headers into list
+			List<WebElement> headers = rows.get(0).findElements(By.tagName("th"));
+			for (WebElement header : headers) {
+				// Extract data from each header and add it to the list
+				if (count == 0) {
+					rowData = header.getText();
+				} else {
+					rowData = rowData + "," + header.getText();
+				}
+				count++;
+			}
+			tableData.add(rowData);
+
+			//Get data of table rows
+			for (int i = 1; i < rows.size(); i++) {
+				// Get all cells in the current row
+				count=0;
+				List<WebElement> cells = rows.get(i).findElements(By.tagName("td"));
+				for (WebElement cell : cells) {
+					// Extract data from each cell and add it to the list
+					if (count == 0) {
+						rowData = cell.getText();
+					} else {
+						rowData = rowData + "," + cell.getText();
+					}
+					count++;
+				}
+				tableData.add(rowData);
+			}
+		}
+		catch (Exception e) {
+			WebUI.comment("Exception :: " + e.getMessage())
+			KeywordUtil.markFailedAndStop("Exception while reading data from WebTable");
+		}
+		return tableData;
+	}
+
+	@Keyword(keywordObject = "WebTable")
+	/**
+	 * Get Web table cell value
+	 * @param webTableObject - WebTable object referenced from OR
+	 * @param primaryKey - primary key of WebTable
+	 * @param primaryKeyValue - primary key value of row
+	 * @param columnName  - column to fetch cell value
+	 */
+	public static String getWebTableCellValue(TestObject webTableObject,String primaryKey,String rowName,String columnName) {
+		try {
+			String cellValue;
+			Map<String, Map<String, String>> webTableMap = getWebTableAsMap(webTableObject, primaryKey);
+			//Iterating through Webtable maps
+			for (Entry<String, Map<String, String>> entry : webTableMap.entrySet()) {
+				if (entry.getKey().equals(rowName)) {
+					cellValue = entry.getValue().get(columnName);
+					break; //Exit inner loop once found
+				}
+			}
+			if (cellValue != null) {
+				KeywordUtil.markPassed("Cell Value for row name: '" + rowName + "' and column name: '" + columnName+"' is '"+cellValue+"'")
+			} else {
+				KeywordUtil.markFailedAndStop("Cell with row name '" + rowName + "' and column name '" + columnName + "' not found in the table with primarykey: '"+primaryKey+"'")
+			}
+			return cellValue;
+		}
+		catch (Exception e) {
+			WebUI.comment("Exception :: " + e.getMessage())
+			KeywordUtil.markFailedAndStop("Exception while fetching cell value from WebTable");
+		}
+	}
+	/**
+	 * to get the read only cells
+	 * @param tableTitle - Title of the table to locate
+	 * @param disabled   - The class that indicates a readOnly state
+	 *
+	 */
+	@Keyword(keywordObject = 'readOnly')
+	public static void verifyTableCellIsEnabled(String tableTitle, String disabled) {

```

Comment:
We are validating that webtable cell is disabled or not but your function name is "verifyTableCellIsEnabled". Is it not contradictory?

Reply:
Done